src/helpers/data.js

```diff
@@ -248,3 +248,35 @@ export function countFirstRowKeys(data) {
 
   return result;
 }
+
+/**
+ * Check whether the provided dataset is an array of arrays.
+ *
+ * @param {Array} data Dataset to be checked.
+ * @returns {boolean} `true` if data is an array of arrays, `false` otherwise.
+ */
+export function isAoA(data) {
+  return !!(
+    Array.isArray(data) &&
+    data.length &&
+    !data.some(el => !Array.isArray(el))
+  );
+}
+
+/**
+ * Check whether the provided dataset is an array of objects.
+ *
+ * @param {Array} data Dataset to be checked.
+ * @returns {boolean} `true` if data is an array of objects, `false` otherwise.
+ */
+export function isAoO(data) {
+  return !!(
+    Array.isArray(data) &&
+    data.length &&
+    !data.some(
+      el =>
+        typeof el !== 'object' ||
+        Array.isArray(el) ||
+        el === null)
+  );
```

Comment:
This will return true for `[]`, that's why I added the `.length` condition.

Reply:
I think "array of arrays" is pretty specific, would you say that `[]` is an array of arrays _and_ an array of objects?

Reply:
Also true! Shame on me for not thinking this through 🙈

Reply:
> I think "array of arrays" is pretty specific, would you say that [] is an array of arrays and an array of objects?

Sounds counter intuitive but yes

Reply:
I'd say that `[]` is not an array of anything, as it's an array of nothing 🤔

Reply:
Anyway, I get your point, it's a little like a _schrodinger's array_, but in the context of HOT helpers, I think it's pretty clear what it's supposed to check.

Reply:
No, you can't put "nothing" inside an array, but you can call it an empty array. Just following js logic - `[].every(x => typeof x === 'number')` is `true` therefore `[]` is an array of numbers, similarly in typescript `[]` could be assigned to both `number[]` and `string[]` (or in our case, array of arrays or array of objects).

I think whether you agree with the above or not we could at least remove the ambiguity from the name `isArrayOfArrays` to include the non-empty condition, as right now as you can see it could mean both depending on how you look at it.

Reply:
You're technically correct, but I'd stick with the simpler naming in this context. I mean, should the helper `isNumeric` accept `null` as a number, because `isNaN(null) => false`, or `isObject` accept `[]` as an object, because `typeof [] => 'object'`? I'd say if someone's checking _specifically_ if an array is an array of objects, it'd better have some objects in it ;PP

Maybe we can meet halfway and change the description to this?
```
Check whether the provided dataset is a *non empty* array of arrays.
```

Reply:
Sure, why not